minifi-api/src/main/java/org/apache/nifi/minifi/configuration/ConfigurationChangeListener.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.minifi.configuration;
+
+/**
+ * Interface for handling events detected and driven by an associated {@link ConfigurationChangeNotifier} to which the listener
+ * has registered via {@link ConfigurationChangeNotifier#registerListener(ConfigurationChangeListener)}.
+ */
+public interface ConfigurationChangeListener {
+
+    /**
+     * Provides a mechanism for the implementation to interpret the specified
+     */
+    void handleChange();
```

Comment:
Fair point on this one.  I was thinking a bit narrowly on the file case, but providing an InputStream to the consumer (in this case bootstrap) should provide some nice flexibility on both sides of the process.